### ISSUE TITLE
[bug] why am i stuck on rn-video 5.2.1 not expo-av or rnv6

### ISSUE THREAD

**Issue by lovegaoshi** [lovegaoshi/azusa-player-mobile#258]
{"error":{"errorStackTrace":"com.google.android.exoplayer2.ExoPlaybackException: Source error\n\tat com.google.android.exoplayer2.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:684)\n\tat com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:656)\n\tat android.os.Handler.dispatchMessage(Handler.java:102)\n\tat android.os.Looper.loopOnce(Looper.java:205)\n\tat android.os.Looper.loop(Looper.java:294)\n\tat android.os.HandlerThread.run(HandlerThread.java:67)\nCaused by: com.google.android.exoplayer2.upstream.HttpDataSource$InvalidResponseCodeException: Response code: 403\n\tat com.google.android.exoplayer2.ext.okhttp.OkHttpDataSource.open(OkHttpDataSource.java:329)\n\tat com.google.android.exoplayer2.upstream.DefaultDataSource.open(DefaultDataSource.java:269)\n\tat com.google.android.exoplayer2.upstream.StatsDataSource.open(StatsDataSource.java:90)\n\tat com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:1013)\n\tat com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:420)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)\n\tat java.lang.Thread.run(Thread.java:1012)\n","errorCode":"22004","errorException":"com.google.android.exoplayer2.ExoPlaybackException: Source error","errorString":"ExoPlaybackException: ERROR_CODE_IO_BAD_HTTP_STATUS"}}

basically header is not accepted upon changing src i think

**Comment by lovegaoshi** [comment]
```
METHOD: GET
URL
https://upos-sz-mirroraliov.bilivideo.com/upgcxcode/30/43/1124324330/1124324330_nb3-1-30080.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1702936104&gen=playurlv2&os=aliovbv&oi=1185180348&trid=f814d3d1a324436bb35e143fc9265429u&mid=0&platform=pc&upsig=0a26a630bcdb0129194b7c50d13dff93&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&bvc=vod&nettype=0&orderid=0,1&buvid=&build=0&f=u_0_0&agrr=1&bw=28250&logo=80000000
HEADERS
Accept-Encoding:
identity
Connection:
Keep-Alive
Host:
upos-sz-mirroraliov.bilivideo.com
Icy-MetaData:
1
referer:
https://www.bilibili.com/
User-Agent:
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36 Edg/118.0.2088.46
User-Agent:
ReactNativeVideo/2.12.5 (Linux;Android 14) AndroidXMedia3/1.1.1
```

**Comment by lovegaoshi** [comment]
5.2.1:
```
METHOD: GET
URL
https://upos-hz-mirrorakam.akamaized.net/upgcxcode/30/43/1124324330/1124324330_nb3-1-30080.m4s?e=ig8euxZM2rNcNbdlhoNvNC8BqJIzNbfqXBvEqxTEto8BTrNvN0GvT90W5JZMkX_YN0MvXg8gNEV4NC8xNEV4N03eN0B5tZlqNxTEto8BTrNvNeZVuJ10Kj_g2UB02J0mN0B5tZlqNCNEto8BTrNvNC7MTX502C8f2jmMQJ6mqF2fka1mqx6gqj0eN0B599M=&uipk=5&nbs=1&deadline=1702936442&gen=playurlv2&os=akam&oi=1185180348&trid=d5014af2fd704a5ab347356e3e39b815u&mid=0&platform=pc&upsig=c734d18f99c8e2844b526468d63f1121&uparams=e,uipk,nbs,deadline,gen,os,oi,trid,mid,platform&hdnts=exp=1702936442~hmac=816a2179c465cb36dbc56b2b2b3436036c1969eae816d9f0982810acbe68bd40&bvc=vod&nettype=0&orderid=0,1&buvid=&build=0&f=u_0_0&agrr=1&bw=28250&logo=80000000
HEADERS
Accept-Encoding:
gzip
Connection:
Keep-Alive
Host:
upos-hz-mirrorakam.akamaized.net
referer:
https://www.bilibili.com/
User-Agent:
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36 Edg/118.0.2088.46
```

**Comment by lovegaoshi** [comment]
![Screenshot from 2023-12-18 12-43-32](https://github.com/lovegaoshi/azusa-player-mobile/assets/106490582/c715bc61-4f3a-4a4b-8972-f3a9d6bbaac6)
video on loop also loops the source get request
